accounting/invoice/views.py

```diff
@@ -20,3 +20,44 @@
 """
 Invoice views.
 """
+
+from rest_framework import decorators, permissions, response, viewsets
+
+from accounting.invoice import choices, models
+
+
+class InvoiceViewSet(viewsets.ModelViewSet):
+    """
+    A view for retrieving and updating `Invoice` objects.
+    """
+
+    permission_classes = (permissions.IsAdminUser,)
+    queryset = models.Invoice.objects.all()
+    lookup_field = 'uuid'
+
+    @decorators.detail_route(
+        permission_classes=[permissions.AllowAny],
+        queryset=models.Invoice.objects.filter(approved=choices.InvoiceApproval.not_approved)
+    )
+    def approve(self, request, uuid=None):
+        """
+        Approve the invoice with UUID `uuid`.
+        """
+        invoice = self.get_object()
+        invoice.approved = choices.InvoiceApproval.manually
+        invoice.save()
+        return response.Response({'approved': True})
+
+    @decorators.detail_route(queryset=models.Invoice.objects.filter(paid=False))
+    def pay(self, request, uuid=None):
```

Comment:
@UmanShahzad Where is the `pay` view linked from? The invoices are marked as paid from the admin, no? If so why do we need this view?

Reply:
I realized when writing the view above it that a similar `pay` one would be really fast to do, and is going to be necessary when the UI comes out. So I just did it. Let's not erase that, it's not going to bite anyone.

Reply:
@UmanShahzad I'm not really sure we'll ever have a need for this method? I might be missing it, but I don't see a corresponding user action, since the admin would always be the person marking the invoices as paid?

But that's fine to leave. Just know that on my side, I love to delete code that's not being used :) Even if it is harmless, it's still code we have to maintain - generally, as you have noticed, I would prefer to keep the code simpler and closer to the use case. I've been bitten too many times by code that was implementing features that users didn't actually need, and were sinking time an money that should have gone toward simpler features that users actually really needed - so that's often a large part of my focus in reviews.

Reply:
@antoviaque It's a little far-fetched but I can imagine the front-end admin UI having the ability to mark the invoices as paid directly without having to go into Django, and for that we need an endpoint. But I get what you mean -- noted.